src/modm/architecture/interface/can_message.hpp.in

```diff
@@ -21,22 +22,21 @@
 namespace modm::can
 {
 
-/// Representation of a CAN message
 /// @ingroup modm_architecture_can
 struct Message
 {
 	inline Message(uint32_t inIdentifier = 0, uint8_t inLength = 0) :
-		identifier(inIdentifier), flags(), length(inLength)
+		identifier(inIdentifier), flags()
 	{
+        setLength(inLength);
 	}
 
-	// Create CAN message from long data in Network Order.
-	inline Message(uint32_t inIdentifier, uint8_t inLength, const uint64_t &inData, bool extended=false) :
-		identifier(inIdentifier), length(std::min(inLength, uint8_t(8)))
+	inline Message(uint32_t inIdentifier, uint8_t inLength, const uint64_t &inData, bool extended = false) :
+		Message(inIdentifier, inLength)
```

Comment:
Before the change specifying `inLength > 8` truncated the message length to 8. Now this will result in out-of-bounds memory accesses. Is this intended to work somehow?

Just as a side-note: Messages created using this constructor have a 11-bit standard id by default, the other constructor uses 29 bit extended. I would not change it because that could break perfectly fine user code at runtime.

Reply:
I have changed the code to the previous state, where the in length is truncated at 8 and added a new constructor